### PR TITLE
InsightIDR Enable User in Active Directory

### PR DESCRIPTION
**Pull request by lmilby-r7**
## Requirements

Developers, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow/Snippet contains title and detailed description
- [ ] Workflow/Snippet title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] User friendly step names in [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] User friendly step descriptions where needed
- [ ] Workflow contains at least 1 artifact with well formatted information
- [ ] Directory created and named after the title the snippet or workflow e.g. `workflows/Basic_Domain_Enrichment_Report/`, `snippets/Domain_Enrichment`
- [ ] Workflow `.icon` file is in the workflow's directory e.g. `workflows/Basic_Domain_Enrichment_Report/Basic_Domain_Enrichment_Report.icon`
- [ ] Take PNG screenshots of the workflow or snippet e.g.  https://market.komand.com/workflows/ross/chatops-github-commands/1.0.0, https://market.komand.com/snippets/jschipp/basic-domain-enrichment-report/0.1.0
- [ ] Create a directory called `screenshots` in your workflow's/snippet's directory
- [ ] Includes PNG screenshots of the workflow in canvas, job, artifact in the `screenshots` directory e.g.  `workflows/Basic_Domain_Enrichment_Report/screenshots/`
- [ ] PNG screenshot file names should conform to style e.g. `workflow1.png`, `artifact1.png`, `job.png`
- [ ] Workflow verified as working through WUI
- [ ] Credentials are not exported and stored in git
- [ ] `workflow.spec.yaml` file with metadata accompanies workflow
- [ ] Workflow documentation is provided in `help.md`
- [ ] Directory structure is followed, see https://github.com/rapid7/insightconnect-workflows/blob/master/README.md

### Assessment

Post your results with screenshots in this PR using the following style:

```
### UI

Screenshots of the workflow being used in Komand for validating use (drag and drop screenshots).

#### Workflow Builder

<img width="697" alt="statsd_workflow" src="https://user-images.githubusercontent.com/2321183/27349935-8e00fd12-55be-11e7-9d86-31250743dc32.png">

#### Job

<img width="1022" alt="screen shot 2017-06-20 at 2 03 31 pm" src="https://user-images.githubusercontent.com/2321183/27350999-9ef49fe0-55c1-11e7-9a74-7b31927d4d9a.png">

#### Artifact

<img width="937" alt="screen shot 2017-06-20 at 2 03 41 pm" src="https://user-images.githubusercontent.com/2321183/27351011-a4f41880-55c1-11e7-89fa-0e671221863e.png">
```

### CODE REVIEW

File: workflows/InsightIDR_Enable_User_in_Active_Directory/help.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Description
+
+This workflow enable a user in Active Directory from an alert sent by InsightIDR
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/help.md]:
```suggestion
This workflow enables a user in Active Directory from an alert sent by InsightIDR.
```

File: workflows/InsightIDR_Enable_User_in_Active_Directory/help.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# Key Features
+
+* Enable user based on an alert from InsightIDR
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/help.md]:
```suggestion
* User provisioning 
```

File: workflows/InsightIDR_Enable_User_in_Active_Directory/help.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+This workflow enable a user in Active Directory from an alert sent by InsightIDR
+
+# Key Features
+
+* Enable user based on an alert from InsightIDR 
+
+# Requirements
+
+Account credentials for an Active Directory user
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/help.md]: Bullets are used in this section. Also, you don't need admin for InsightConnect

```suggestion
* Account credentials for an Active Directory user
* Admin access to Rapid7 InsightIDR and Rapid7 InsightConnect
```

File: workflows/InsightIDR_Enable_User_in_Active_Directory/help.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+_There is no troubleshooting information at this time_
+
+# Version History
+
+* 1.0.0 - Initial workflow
+
+# Links
+
+## Source Code
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/help.md]: We need to remove the Source Code section from all workflow files.

it's now presented in the spec as, per the hackathon overview call this morning

```
resources:
  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/<directory>
```

File: workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+- ldap
+- active_directory
+- enable
+- user_management
+hub_tags:
+  use_cases: [user_management]
+  keywords: [insightidr, ldap, ad, active_directory]
+  features: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-workflows
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml]: Use full path now for this, adjusted during hackathon call today

File: workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon [hunk 6]
```diff
@@ -0,0 +1,761 @@
...
+  "kom": {
+    "komFileVersion": "",
+    "komandVersion": "",
+    "exportedAt": "2019-06-20T16:31:01.274574328Z",
+    "workflowVersions": [
+      {
+        "id": "df71f96a-047b-4283-9241-dd1ff374c281",
+        "workflowId": "9a50374d-80ac-45f7-8a11-f7d1077e4eba",
+        "version": "",
+        "name": "Enable User in Active Directory",
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon]: I think spec['title'] is supposed to generate this. Looks out of date.

File: workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon [hunk 7]
```diff
@@ -0,0 +1,761 @@
+{
+  "kom": {
+    "komFileVersion": "",
+    "komandVersion": "",
+    "exportedAt": "2019-06-20T16:31:01.274574328Z",
+    "workflowVersions": [
+      {
+        "id": "df71f96a-047b-4283-9241-dd1ff374c281",
+        "workflowId": "9a50374d-80ac-45f7-8a11-f7d1077e4eba",
+        "version": "",
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon]: Set 1.0.0 from spec, tooling supposed to gen this

File: workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon (hunk 6 above)
Comment (lmilby-r7) [reply workflows/InsightIDR_Enable_User_in_Active_Directory/InsightIDR_Enable_User_in_Active_Directory.icon]: tooling needs more work.

File: workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+extension: workflow
+product: ["insightconnect"]
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml]:
```suggestion
product: [insightconnect, insightidr]
```

File: workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+support: rapid7
+status: []
+tags:
+- insightidr
+- ldap
+- active_directory
+- enable
+- user_management
+hub_tags:
+  use_cases: [user_management]
```
Comment (jschipp-r7) [inline workflows/InsightIDR_Enable_User_in_Active_Directory/workflow.spec.yaml]: Add 

```suggestion
  use_cases: [user_management, threat_detection_and_response]
```

**Pull request merged**